### PR TITLE
Document AsyncLocalStorage support

### PR DESCRIPTION
**Pull request by jasnell**
This is the start of documentation for the `AsyncLocalStorage` implementation in workers.

### CODE REVIEW

**Comment by jasnell** [comment]
@deadlypants1973 ... this is ready to go!

**Review by deadlypants1973 — APPROVED** [review]

File: content/workers/runtime-apis/nodejs/asynclocalstorage.md [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+
+{{</definitions>}}
+
+## Methods
+
+{{<definitions>}}
+
+- `getStore()` {{<type>}}any{{</type>}}
+
+  - Returns the current stored. If called outside of an asynchronous context initialized by calling `asyncLocalStorage.run()`, it returns `undefined`.
```
Comment (irvinebroque) [inline content/workers/runtime-apis/nodejs/asynclocalstorage.md:36]:
```suggestion
  - Returns the current store. If called outside of an asynchronous context initialized by calling `asyncLocalStorage.run()`, it returns `undefined`.
```

File: content/workers/runtime-apis/nodejs/asynclocalstorage.md [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+      await scheduler.wait(1000);
+      return new Response(asyncLocalStorage.getStore());
+    });
+  }
+};
+```
+
+### Multiple stores
+
+The API supports multiple `AsyncLocalStorage` instances to be used concurrently.
```
Comment (irvinebroque) [inline content/workers/runtime-apis/nodejs/asynclocalstorage.md:85]:
```suggestion
You can use multiple `AsyncLocalStorage` instances concurrently:
```

File: content/workers/runtime-apis/nodejs/asynclocalstorage.md [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+
+const myResource = als.run(123, () => new MyResource());
+console.log(myResource.doSomething()); // prints 123
+```
+
+## AsyncResource
+
+The [`AsyncResource`](https://nodejs.org/dist/latest-v18.x/docs/api/async_context.html#class-asyncresource) class is a component of Node.js' async context tracking API that allows users to create their own async contexts. Objects that extend from `AsyncResource` are capable of propagating the async context in much the same way as promises.
+
+Note that `AsyncLocalStorage.snapshot()` and `AsyncLocalStorage.bind()` provide a better approach. `AsyncResource` is provided solely for backwards compatibility with Node.js.
```
Comment (irvinebroque) [inline content/workers/runtime-apis/nodejs/asynclocalstorage.md:171]:
```suggestion
Note that the same thing can be accomplished using `AsyncLocalStorage.snapshot()` and `AsyncLocalStorage.bind()`, and this approach is recommended. `AsyncResource` is provided solely for backwards compatibility with Node.js, and is not currently part of the [TC39 proposal for `AsyncContext`](https://github.com/tc39/proposal-async-context).
```

**Review by irvinebroque — APPROVED** [review]
🚀 a few small comments, feel free to take or leave. Wonder if it's worthwhile to link to the `AsyncContext` proposal?

**Pull request merged**